crates/ruff_server/src/trace.rs

```diff
@@ -43,13 +49,46 @@ pub(crate) fn set_trace_value(trace_value: TraceValue) {
     *global_trace_value = trace_value;
 }
 
+// A tracing writer that uses LSPs logTrace method.
+struct TraceLogWriter;
+
+impl Write for TraceLogWriter {
+    fn write(&mut self, buf: &[u8]) -> std::io::Result<usize> {
+        let message = str::from_utf8(buf).map_err(|e| IoError::new(ErrorKind::InvalidData, e))?;
+        LOGGING_SENDER
+            .get()
+            .expect("logging sender should be initialized at this point")
+            .send(Message::Notification(Notification {
+                method: LogTrace::METHOD.to_owned(),
+                params: serde_json::json!({
+                    "message": message
+                }),
+            }))
+            .map_err(|e| IoError::new(ErrorKind::Other, e))?;
+        Ok(buf.len())
+    }
+
+    fn flush(&mut self) -> std::io::Result<()> {
+        Ok(())
+    }
+}
+
+impl<'a> MakeWriter<'a> for TraceLogWriter {
+    type Writer = Self;
+
+    fn make_writer(&'a self) -> Self::Writer {
+        Self
+    }
+}
+
 pub(crate) fn init_tracing(
-    sender: ClientSender,
+    sender: &ClientSender,
     log_level: LogLevel,
     log_file: Option<&std::path::Path>,
+    client: &Option<ClientInfo>,
 ) {
     LOGGING_SENDER
-        .set(sender)
+        .set(sender.clone())
```

Comment:
Any specific reason why `ClientSender` is a reference? We can just pass in the value as it's going to be consumed anyway.

Reply:
Clippy was just complaining about it in [CI](https://github.com/astral-sh/ruff/actions/runs/10142100234/job/28040714091), no other reason.